src/python/WMCore/DataStructs/Run.py

```diff
@@ -157,13 +157,20 @@ def extendLumis(self, lumiList):
         Method to replace myRun.lumis.extend() which does not work with the property
         """
         for lumi in lumiList:
-            if isinstance(lumi, (list, tuple)) and lumi[0] in self.eventsPerLumi and self.eventsPerLumi[lumi[0]]:
-                self.eventsPerLumi[lumi[0]] += lumi[1]  # Already exists, add events
-            elif isinstance(lumi, (list, tuple)):  # Doesn't exist or is 0 or None
-                self.eventsPerLumi[lumi[0]] = lumi[1]
-            else:  # Just given lumis, not events
-                if lumi not in self.eventsPerLumi:  # Don't overwrite existing events
-                    self.eventsPerLumi[lumi] = None
+            if not isinstance(lumi, (list, tuple)):  # comma separated lumi numbers
+                self.eventsPerLumi[lumi] = None
+            else:
+                if isinstance(lumi, list) and not isinstance(lumi[0], tuple):  # then it's a plain list
+                    self.eventsPerLumi = {l: None for l in lumi}
```

Comment:
This dictionary comprehension can crash the job during runtime IF the CMSSW uses python2.6. Posting more info outside of the code review.